tiled/server/core.py

```diff
@@ -367,16 +372,21 @@ def construct_dataframe_response(
                 ","
             )
         ]
+
+    # fall back to generic dataframe serializer if no specs present
+    specs.append("dataframe")
+
     # The client may give us a choice of media types. Find the first one
     # that we support.
     for media_type in media_types:
         if media_type == "*/*":
             media_type = APACHE_ARROW_FILE_MIME_TYPE
-        if media_type in serialization_registry.media_types("dataframe"):
-            content = serialization_registry("dataframe", media_type, df, metadata)
-            return PatchedResponse(
-                content=content, media_type=media_type, headers={"ETag": etag}
-            )
+        for spec in specs:
```

Comment:
Can we do `spec in specs + ["dataframe"]` rather than mutating `specs` itself above? Mutating the user input could have confusing side-effects.